reana_db/models.py

```diff
@@ -1151,3 +1155,28 @@ class QuotaHealth(enum.Enum):
     healthy = 0
     warning = 1
     critical = 2
+
+
+class AccessType(enum.Enum):
+    """Enumeration of access types."""
+
+    read = 0
+
+
+class UserWorkflow(Base):
+    """UserWorkflow table."""
+
+    __tablename__ = "user_workflow"
+    __table_args__ = {"schema": "__reana"}
+
+    workflow_id = Column(UUIDType, ForeignKey("__reana.workflow.id_"), primary_key=True)
+    user_id = Column(UUIDType, ForeignKey("__reana.user_.id_"), primary_key=True)
+    access_type = Column(Enum(AccessType), default=AccessType.read, nullable=False)
+    message = Column(Text)
```

Comment:
minor: we could consider limiting the length of this field also in the database schema by using `String(a1234...)`, up to you!